File: app/src/main/java/com/pij/zworkout/service/android/Threader.java
package com.pij.zworkout.service.android;

import android.support.annotation.NonNull;

import io.reactivex.CompletableTransformer;
import io.reactivex.ObservableTransformer;
import io.reactivex.SingleTransformer;

/**
 * <p>Created on 16/03/2018.</p>
 *
 * @author Pierrejean
 */

interface Threader {

    @NonNull
    <T> ObservableTransformer<T, T> forObservable();

    @NonNull
    <T> SingleTransformer<T, T> forSingle();

    @NonNull
    CompletableTransformer forCompletable();
}

File: app/src/main/java/com/pij/zworkout/service/android/Threader.kt
package com.pij.zworkout.service.android

import io.reactivex.CompletableTransformer
import io.reactivex.ObservableTransformer
import io.reactivex.SingleTransformer

/**
 *
 * Created on 16/03/2018.
 *
 * @author Pierrejean
 */

internal interface Threader {

    fun <T> forObservable(): ObservableTransformer<T, T>

    fun <T> forSingle(): SingleTransformer<T, T>

    fun forCompletable(): CompletableTransformer
}
